src/Sulu/Bundle/ContentBundle/Controller/NodeController.php

```diff
@@ -590,14 +591,18 @@ public function filterAction(Request $request)
     public function putAction(Request $request, $uuid)
     {
         $language = $this->getLanguage($request);
-        $type = $request->query->get('type', 'page');
 
         $document = $this->getDocumentManager()->find(
             $uuid,
             $language,
-            ['load_ghost_content' => false]
+            [
+                'load_ghost_content' => false,
+                'load_shadow_content' => false,
+            ]
         );
 
+        $type = $this->getMetadataFactory()->getMetadataForClass(get_class($document))->getAlias();
```

Comment:
@chirimoya I have changed the type detection to work on the document, because the `type` parameter has not to be passed via the request then. Also removed the option from the `postAction`, since this Controller is only creating documents with the type `page`. Only exception is this line, since this controller also edits documents of type `home`, but it does not create them.